File: Permuting-Two-Arrays.ts
'use strict';

import { WriteStream, createWriteStream } from "fs";
process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString: string = '';
let inputLines: string[] = [];
let currentLine: number = 0;

process.stdin.on('data', function(inputStdin: string): void {
    inputString += inputStdin;
});

process.stdin.on('end', function(): void {
    inputLines = inputString.split('\n');
    inputString = '';

    main();
});

function readLine(): string {
    return inputLines[currentLine++];
}

/*
 * Complete the 'twoArrays' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts following parameters:
 *  1. INTEGER k
 *  2. INTEGER_ARRAY A
 *  3. INTEGER_ARRAY B
 */

function twoArrays(k: number, A: number[], B: number[]): string {
    // Sort A in ascending order
    A.sort((a, b) => a - b);

    // Sort B in descending order
    B.sort((a, b) => b - a);

    // Check if every pair sum is at least k
    for (let i = 0; i < A.length; i++) {

        if (A[i] + B[i] < k) {
            return "NO";
        }
    }

    return "YES";    
}

File: Permuting-Two-Arrays.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'twoArrays' function below.
#
# The function is expected to return a STRING.
# The function accepts following parameters:
#  1. INTEGER k
#  2. INTEGER_ARRAY A
#  3. INTEGER_ARRAY B
#

def twoArrays(k, A, B):
    # Sort A in ascending order
    A.sort()

    # Sort B in descending order
    B.sort(reverse=True)

    # Check if every pair sum is at least k
    for i in range(len(A)):
        if A[i] + B[i] < k:
            return "NO"

    return "YES"    

    
